libs/sdk-ui-dashboard/src/presentation/filterBar/filterBar/filterViews/FilterViews.tsx

```diff
@@ -1,64 +1,141 @@
 
-import React from "react";
+import React, { useCallback } from "react";
 import { FormattedMessage } from "react-intl";
 import cx from "classnames";
-import { IAlignPoint, DropdownButton } from "@gooddata/sdk-ui-kit";
+import { IAlignPoint, DropdownButton, useMediaQuery } from "@gooddata/sdk-ui-kit";
+import { IDashboardFilterView } from "@gooddata/sdk-model";
 
 import { ConfigurationBubble } from "../../../widget/common/configuration/ConfigurationBubble.js";
-import { useDashboardSelector, selectFilterViews, selectIsInEditMode } from "../../../../model/index.js";
+import {
+    useDashboardSelector,
+    selectFilterViews,
+    selectIsInEditMode,
+    useDashboardDispatch,
+    uiActions,
+    selectDisableFilterViews,
+} from "../../../../model/index.js";
+import {
+    selectIsFilterViewsDialogOpen,
+    selectFilterViewsDialogMode,
+} from "../../../../model/store/ui/uiSelectors.js";
 
 import { FilterViewsList } from "./FilterViewsList.js";
 import { useFilterViewsToastMessages } from "./useFilterViewsToastMessages.js";
 import { AddFilterView } from "./AddFilterView.js";
 
 const BUBBLE_ALIGN_POINTS: IAlignPoint[] = [{ align: "br tr", offset: { x: -27, y: -10 } }];
 
-export const FilterViews: React.FC = () => {
-    const [isOpen, setIsOpen] = React.useState(false);
-    const onClose = () => setIsOpen(false);
-    const [isAddNew, setAddNew] = React.useState(false);
-    const isEditMode = useDashboardSelector(selectIsInEditMode);
+const DropdownButtonLabel: React.FC<{ filterViews: IDashboardFilterView[] }> = ({ filterViews }) => {
+    return filterViews.length === 0 ? (
+        <FormattedMessage id="filters.filterViews.dropdown.buttonEmpty" />
+    ) : (
+        <FormattedMessage
+            id="filters.filterViews.dropdown.button"
+            values={{ count: filterViews?.length ?? 0 }}
+        />
+    );
+};
+
+const useCallbacks = () => {
+    const dispatch = useDashboardDispatch();
+
+    const toggleDialog = useCallback(() => dispatch(uiActions.toggleFilterViewsDialog()), [dispatch]);
+
+    const openListDialog = useCallback(
+        () =>
+            dispatch(
+                uiActions.toggleFilterViewsDialog({
+                    open: true,
+                    mode: "list",
+                }),
+            ),
+        [dispatch],
+    );
+
+    const openAddDialog = useCallback(
+        () =>
+            dispatch(
+                uiActions.toggleFilterViewsDialog({
+                    open: true,
+                    mode: "add",
+                }),
+            ),
+        [dispatch],
+    );
+
+    const closeDialog = useCallback(
+        () =>
+            dispatch(
+                uiActions.toggleFilterViewsDialog({
+                    open: false,
+                }),
+            ),
+        [dispatch],
+    );
+
+    return {
+        toggleDialog,
+        openListDialog,
+        openAddDialog,
+        closeDialog,
+    };
+};
 
+export const FilterViews: React.FC = () => {
```

Comment:
Component open / mode state is in redux now so component can be managed from mobile menu dropdown.